holoviews/plotting/bokeh/util.py

```diff
@@ -226,15 +242,14 @@ def delete_refs(obj, locs, delete):
             return None
         new_obj = {}
         for k, v in list(obj.items()):
-            if k in locs:
-                ref = delete_refs(v, locs, delete)
-                if ref is not None:
-                    new_obj[k] = ref
-            else:
-              new_obj[k] = v
+            if k in ['data', 'palette'] or (k == 'attributes' and not get_ids(v)):
```

Comment:
Will this list of hardcoded strings potentially change? What do they correspond to? It might be worth specifying them as an optional argument (tuple).